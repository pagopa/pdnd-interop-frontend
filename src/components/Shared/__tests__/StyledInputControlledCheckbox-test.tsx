import React from 'react'
import { useForm } from 'react-hook-form'
import renderer from 'react-test-renderer'
import { StyledInputControlledCheckbox } from '../StyledInputControlledCheckbox'

type Option = {
  value: string
  label: string
}

type CheckboxProps = {
  options?: Array<Option>
}

function Checkbox({ options }: CheckboxProps) {
  const props = { name: 'test', rules: {} }

  const {
    control,
    formState: { errors },
  } = useForm()

  return (
    <StyledInputControlledCheckbox {...props} control={control} errors={errors} options={options} />
  )
}

describe('Snapshot', () => {
  it('matches null', () => {
    const component = renderer.create(<Checkbox />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches options', () => {
    const options = [
      { label: 'Prima opzione', value: 'first' },
      { label: 'Seconda opzione', value: 'second' },
    ]
    const component = renderer.create(<Checkbox options={options} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
