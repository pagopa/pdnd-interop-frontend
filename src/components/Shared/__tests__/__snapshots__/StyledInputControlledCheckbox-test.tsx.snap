// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches null 1`] = `null`;

exports[`Snapshot matches options 1`] = `
<div
  className="MuiBox-root css-62igne"
>
  <div
    className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="first"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Prima opzione
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
          value="second"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Seconda opzione
      </span>
    </label>
  </div>
</div>
`;
